driver-core/src/main/com/mongodb/client/model/expressions/IntegerExpression.java

```diff
@@ -19,32 +19,98 @@
 import java.util.function.Function;
 
 /**
- * Expresses an integer value.
+ * An integer value. Integers are a subset of {@link NumberExpression numbers},
+ * and so, for example, the integer 0 and the number 0 are the same value,
+ * and are equal.
  */
 public interface IntegerExpression extends NumberExpression {
-    IntegerExpression multiply(IntegerExpression i);
 
-    default IntegerExpression multiply(final int multiply) {
-        return this.multiply(Expressions.of(multiply));
+    /**
+     * The product of multiplying {@code this} and the {@code other} value.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    IntegerExpression multiply(IntegerExpression other);
+
+    /**
+     * The product of multiplying {@code this} and the {@code other} value.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    default IntegerExpression multiply(final int other) {
+        return this.multiply(Expressions.of(other));
     }
 
-    IntegerExpression add(IntegerExpression i);
+    /**
+     * The sum of adding {@code this} and the {@code other} value.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    IntegerExpression add(IntegerExpression other);
 
-    default IntegerExpression add(final int add) {
-        return this.add(Expressions.of(add));
+    /**
+     * The sum of adding {@code this} and the {@code other} value.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    default IntegerExpression add(final int other) {
+        return this.add(Expressions.of(other));
     }
 
-    IntegerExpression subtract(IntegerExpression i);
+    /**
+     * The result of subtracting the {@code other} value from {@code this}.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    IntegerExpression subtract(IntegerExpression other);
 
-    default IntegerExpression subtract(final int subtract) {
-        return this.subtract(Expressions.of(subtract));
+    /**
+     * The result of subtracting the {@code other} value from {@code this}.
+     *
+     * @param other the other value.
+     * @return the resulting value.
+     */
+    default IntegerExpression subtract(final int other) {
+        return this.subtract(Expressions.of(other));
     }
```

Comment:
For subtract, I considered and avoided "difference" because to me that word might (especially in the absence of the word "subtract") imply "how much of a difference is there between 1 and 2?" -> "1" (so the absolute difference). I think I avoided "quotient" because "the quotient of dividing" did not sound grammatical to me. What do you think?

Reply:
> avoided "difference" because to me that word might (especially in the absence of the word "subtract") imply "how much of a difference is there between 1 and 2?" -> "1"

It would be "difference of subtracting" similarly to "product of multiplying".

> "the quotient of dividing" did not sound grammatical to me

But it is grammatical, it is no different from "sum of adding". The reason it feels weird is because people usually don't use this word in conversations.

> What do you think?

I think that we should be consistent: either we call results of these operations by their names, or we refer to  all of them using a generic term.

Reply:
Alright, fair enough. But I think there are places elsewhere where there is not a more-specific term than "result".